woocommerce/woocommerce-ios
# [Issue Refund] Decode and store Order.paymentMethod

closes #3111

# Why
As a continuation of the "Refund ~ PaymentGateway" work, this PR takes care of decoding and storing the payment gateway id when synchronizing orders.

This `ID` will be used later to fetch the full payment gateway to know if it supports automatic money refunds or not. 😇 

# How
- Added `paymentMethodID` to `Networking.Order`
- Added `paymentMethodID` to `Storage.Order` and updated CoreData model to `v37`

# Testing Steps
- Smoke test around orders and check that everything is working as before.

Update release notes:
